### PR TITLE
- added description to method "on"

### PR DESCRIPTION
**Pull request by CreatingTheNewWorld**
- minor changes in EvolvContextImpl.java
- added setTest() to EvolvContextImplTest.java
- added setActiveAndEntryKeyStatesCaseOne to EvolvStoreImplTest.java
- added setActiveAndEntryKeyStatesCaseTwo to EvolvStoreImplTest.java
- added setActiveAndEntryKeyStatesCaseThree to EvolvStoreImplTest.java

### CODE REVIEW

File: android-sdk/src/test/java/ai/evolv/android_sdk/EvolvStoreImplTest.java [hunk 1]
```diff
@@ -160,15 +167,147 @@ public void testClearActiveKeysPrefix() {
...
+
+        evolvContext.setRemoteContext(remoteContext);
+        EvolvStoreImpl evolvStore = new EvolvStoreImpl(evolvConfig, participant, waitForIt);
+
+        EvolvStoreImpl.KeyStates configKeyStates = new EvolvStoreImpl.KeyStates();
+        configKeyStates.experiments = parseRawJsonObject(rawExperiment_two);
+
+        evolvStore.setActiveAndEntryKeyStates(version, evolvContext, config, allocations, configKeyStates);
+
+        Assert.assertEquals(configKeyStates.experiments, expectedResult);
```
Comment (robertsevernsentient) [inline android-sdk/src/test/java/ai/evolv/android_sdk/EvolvStoreImplTest.java:216]: Can you give some details on what functionality this tests? It looks like configKeyStates.experiments and expectedResult are both objects that you define in this test class

Comment (CreatingTheNewWorld) [reply android-sdk/src/test/java/ai/evolv/android_sdk/EvolvStoreImplTest.java:216]: **setActiveAndEntryKeyStates** - the main method that contains the loaded active and entri keys 
in order to test it, I need to check for a match between '_active_' and '_loaded_' and '_entry_' returned by the method and the expected ones. further, using the example of debugging, I will show what is happening:
изображение
изображение

File: android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java [hunk 2]
```diff
@@ -164,18 +164,14 @@ public boolean set(String key, Object value, boolean local) {
...
         ensureInitialized();
-// TODO: 26.07.2021 uncomment and need to test!
-//        for (Map.Entry<String, JsonElement> entry : localContext.entrySet()) {
-//            remoteContext.add(entry.getKey(),entry.getValue());
-//        }
-//
-//        return remoteContext;
-
-        return new JsonObject();
+
```
Comment (robertsevernsentient) [inline android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java:170]: Why are we copying localContext over to remoteContext?

Comment (CreatingTheNewWorld) [reply android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java:165]: we discussed it in the chat 
(need another implementation like a npm deepmerge js)

Comment (robertsevernsentient) [reply android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java:165]: Can you merge the other option, so at least we're not accidentally sending localcontext data to our server?

Comment (CreatingTheNewWorld) [reply android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java:165]: yes of course, in the last commit it was taken into account 

изображение

**Pull request merged**